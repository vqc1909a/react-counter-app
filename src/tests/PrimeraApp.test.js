// import { render } from '@testing-library/react';
import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';


describe("Pruebas en <PrimeraApp />", () => {
  // test("componente PrimeraApp debe incluir el mensaje 'hola perro de mierda'", () => {
  //   const saludo = "hola perro de mierda";
  //   const wrapper = render(<PrimeraApp saludo={saludo} />);
  //   expect(wrapper.getByText(saludo)).toBeInTheDocument();
  // })

  test("debe de mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "hola perro de mierda";
    const wrapper = shallow(<PrimeraApp saludo={saludo}></PrimeraApp>);
    expect(wrapper).toMatchSnapshot();
  })

  test("debe de mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola perriqui";
    const subtituloTest = "Hola perkin";

    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtituloTest}></PrimeraApp>)
    const subtitulo = wrapper.find("p").text().trim();

    expect(subtitulo).toBe(subtituloTest);

  })
})