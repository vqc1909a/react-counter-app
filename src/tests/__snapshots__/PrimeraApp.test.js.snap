// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola perro de mierda
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 2`] = `
<Fragment>
  <h1>
    hola perro de mierda
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
